src/xpath-evaluation.js

```diff
@@ -165,25 +189,10 @@ function createNamespaceResolver(xpathQuery, formElement) {
     if (cachedResolver) {
         return cachedResolver;
     }
-
-    const xpathAST = parseScript(xpathQuery, {}, xmlDocument);
-    let instanceReferences = fxEvaluateXPathToStrings(
-        `descendant::xqx:functionCallExpr
-				[xqx:functionName = "instance"]
-				/xqx:arguments
-				/xqx:stringConstantExpr
-				/xqx:value`,
-        xpathAST,
-        null,
-        {},
-        {
-            namespaceResolver: prefix =>
-                prefix === 'xqx' ? 'http://www.w3.org/2005/XQueryX' : undefined,
-        },
-    );
+	let instanceReferences = findInstanceReferences(xpathQuery);
     if (instanceReferences.length === 0) {
         // No instance functions. Look up further in the hierarchy to see if we can deduce the intended context from there
-        const ancestorComponent = fxEvaluateXPathToFirstNode('ancestor::*[@ref][1]', formElement);
+        const ancestorComponent = formElement.parentNode.closest('[ref]');
```

Comment:
this change causes tests to fail

Reply:
ok, seems it's just the closest versus fxEvaluateXPathToFirstNode - the latter works all other changes applied